src/utils/strbuf/strbuf.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * collectd - src/utils/strbuf/strbuf.h
+ * Copyright (C) 2017       Florian octo Forster
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Authors:
+ *   Florian octo Forster <octo at collectd.org>
+ */
+
+#ifndef UTILS_STRBUF_H
+#define UTILS_STRBUF_H 1
+
+typedef struct {
+  char *ptr;
+  size_t pos;
+  size_t size;
+  _Bool fixed;
+} strbuf_t;
+
+#define STRBUF_NEW                                                             \
+  &(strbuf_t) { .ptr = NULL }
+#define STRBUF_NEW_STATIC(b)                                                   \
+  &(strbuf_t) { .ptr = b, .size = sizeof(b), .fixed = 1 }
+#define STRBUF_DELETE(buf)                                                     \
+  do {                                                                         \
+    if (!buf->fixed) {                                                         \
+      free(buf->ptr);                                                          \
+    }                                                                          \
+    *buf = (strbuf_t){.ptr = NULL};                                            \
+  }
```

Comment:
These macros are public. Should you be testing them as well?

Reply:
Excellent point, thank you! I've added tests and even found an error with the `STRBUF_DESTROY` macro.